File: swift/example/Tests/VanillaDiExampleTests/BarSpy.swift
@testable import VanillaDiExample


class BarSpy: Bar {
    private(set) var calledArgs = [CallArgs]()


    func doSomething(value: Int) {
        self.calledArgs.append(.doSomething(value: value))
    }


    enum CallArgs: Equatable {
        case doSomething(value: Int)
    }
}

File: cs/example/VanillaDIExample.Tests/BarSpy.cs
using System.Collections.Generic;



namespace VanillaDIExample
{
    public class BarSpy : IBar
    {
        public List<int> CalledArgs => calledArgs;
        private List<int> calledArgs = new List<int>();


        public void DoSomething(int value)
        {
            calledArgs.Add(value);
        }
    }
}